src/base/bittorrent/filterparserthread.cpp

```diff
@@ -37,75 +42,133 @@
+            // to parse possible IPv6 literals wrapped in []. These characters can appear in
+            // the "some organization" part too. It isn't worth implementing such complex parsing
+            // for a format that doesn't support IPv6 and there aren't any lists with IPv6 in them.
+            if (const auto colon = line.rfind(':');
+                colon != line.npos)
+                entry.range = line.substr(colon + 1);
+            else
+                entry.range = line;
+        }
+
+        return entry;
+    }
+
+    std::pair<std::string_view, std::string_view> parseRange(std::string_view range)
+    {
+        std::string_view start;
+        std::string_view end;
 
-            return false;
+        if (const auto pos  = range.find('-');
+            pos != range.npos)
+        {
+            start = range.substr(0, pos);
+            end = range.substr(pos + 1);
         }
 
-        lt::address_v4::bytes_type parsed() const
+        return std::make_pair(start, end);
+    }
+
+    std::optional<lt::address> parseIPv4Address(std::string_view IPstr)
+    {
+        lt::address_v4::bytes_type octets = {};
+        unsigned char octetIndex = 0;
+
+        for (const char *lastptr = IPstr.data();
+             (lastptr < (IPstr.data() + IPstr.size())) && (octetIndex < 4)
+             ;)
         {
-            return m_buf;
+            lt::address_v4::bytes_type::value_type octet = 0;
+
+            auto posIt  = std::find_if_not(lastptr, IPstr.data() + IPstr.size(),
+                                          [](unsigned char c){ return std::isspace(c); } );
+            int pos = 0;
+            if (posIt != IPstr.data() + IPstr.size())
+                pos = posIt - lastptr;
+
+            const auto& [ptr, ec] = std::from_chars(lastptr + pos, IPstr.data() + IPstr.size(), octet);
+            if (ec != std::errc())
+                return {};
+
+            octets[octetIndex++] = octet;
+            lastptr = ptr + 1;
         }
 
-    private:
-        lt::address_v4::bytes_type m_buf;
-    };
+        if (octetIndex == 4)
+            return lt::address_v4(octets);
 
-    bool parseIPAddress(const char *data, lt::address &address)
+        return {};
+    }
+
+    std::optional<lt::address> parseIPAddress(std::string_view IPstr)
     {
-        IPv4Parser parser;
         lt::error_code ec;
 
-        if (parser.tryParse(data))
-            address = lt::address_v4(parser.parsed());
-        else
-            address = lt::make_address(data, ec);
+        if (auto addr = parseIPv4Address(IPstr);
+            addr)
+            return addr;
 
-        return !ec;
+        // Possibly an IPv6 address. Caller checks for validity
+        if (auto addr = lt::make_address(IPstr, ec);
+            !ec)
+            return addr;
+
+        return {};
     }
 
-    const int BUFFER_SIZE = 2 * 1024 * 1024; // 2 MiB
-    const int MAX_LOGGED_ERRORS = 5;
+    constexpr int BUFFER_SIZE = 2 * 1024 * 1024; // 2 MiB
```

Comment:
I would reserve `constexpr` for statements that really require compiler to evaluate at compile-time.

Reply:
Why not? If something can be calculated at compile time, let's do it.

Reply:
https://stackoverflow.com/a/28821610 specifically about the "So, why do we need constexpr?" part.
We can doesn't always mean we should... and this is a personal preference, I don't insist that much to be a blocker.

Reply:
> Talking from experience, by reserving `constexpr` usage, it is easier to follow the code to the part that needs compile-time evaluation.
> 
> Side note, actually a lot of keywords such as `noexcept` declares intention of the programmer and IMO whether to use it depends on our use case at hand.

👍

Reply:
>If something can be calculated at compile time, let's do it.

This constant is calculated in compile time regardless of `constexpr` specifier.

Reply:
I feel like this should be calculated at compile time (the intent). So I want to mark it as constexpr.